docs/hd.md

```diff
@@ -0,0 +1,124 @@
+# Cardano SL HD Wallets
+
+## General scheme
+    
+HD wallet has tree-based structure.
+The root of the tree is user's public-secret key pair.
```

Comment:
I'm a bit on the fence whether or not we should say that each node contain such `(pk,sk)` tuple, because technically you can always derive a `pk` from a `sk`. Maybe we should clarify that:

"Although it's always possible to derive a `pk` from a `sk` due to the mathematical properties of ECC (_Elliptic Curves Cryptography_), for illustration purposes we represent each node as always be composed by a `(public key, secret key)` pair".